File: assignment 9/python/programs/6.py
def find_largest_consecutive_subarray(arr):
    max_length = 0
    max_start = 0
    num_set = set(arr)

    for num in arr:
        if num - 1 not in num_set:
            current_num = num
            current_length = 1

            while current_num + 1 in num_set:
                current_num += 1
                current_length += 1

            if current_length > max_length:
                max_length = current_length
                max_start = num

    largest_subarray = [i for i in range(max_start, max_start + max_length)]
    return largest_subarray


# Example usage
arr = [1, 3, 8, 4, 2, 2, 5, 6]
largest_subarray = find_largest_consecutive_subarray(arr)
print("Largest subarray formed by consecutive integers:", largest_subarray)

File: assignment 9/c/programs/6.c
#include <stdio.h>

void findLargestConsecutiveSubarray(int arr[], int size)
{
    int maxLength = 0;  // Length of the largest consecutive subarray
    int startIndex = 0; // Starting index of the largest consecutive subarray
    int endIndex = 0;   // Ending index of the largest consecutive subarray

    int currentLength = 1; // Length of the current consecutive subarray

    for (int i = 1; i < size; i++)
    {
        if (arr[i] == arr[i - 1] + 1)
        {
            currentLength++;
        }
        else
        {
            if (currentLength > maxLength)
            {
                maxLength = currentLength;
                startIndex = i - maxLength;
                endIndex = i - 1;
            }
            currentLength = 1;
        }
    }

    // Check if the last subarray is the largest
    if (currentLength > maxLength)
    {
        maxLength = currentLength;
        startIndex = size - maxLength;
        endIndex = size - 1;
    }

    if (maxLength > 0)
    {
        printf("Largest consecutive subarray is [%d, %d]\n", startIndex, endIndex);
    }
    else
    {
        printf("No consecutive subarray found.\n");
    }
}

int main()
{
    int arr[] = {1, 2, 3, 4, 5, 7, 8, 9, 10};
    int size = sizeof(arr) / sizeof(arr[0]);

    findLargestConsecutiveSubarray(arr, size);

    return 0;
}
